File: Advanced/PageAndRDL/CustomDataProvider/VB.NET/CustomDataProvider/CSVDataProvider/CsvDataProviderFactory.vb
Imports System.Data.Common

''' <summary>
''' Implements <see cref="DataProviderFactory"/> for .NET Framework CSV Data Provider.
''' </summary>
Public Class CsvDataProviderFactory
	Inherits DbProviderFactory

	''' <summary>
	''' Returns a new instance of the <see cref="CsvCommand"/>.
	''' </summary>
	Public Overrides Function CreateCommand() As DbCommand
		Return New CsvCommand()
	End Function

	''' <summary>
	''' Returns a new instance of the <see cref="CsvConnection"/>.
	''' </summary>
	Public Overrides Function CreateConnection() As DbConnection
		Return New CsvConnection()
	End Function
End Class

File: Advanced/PageAndRDL/CustomDataProvider/C#/CustomDataProvider/CSVDataProvider/CsvDataProviderFactory.cs
using GrapeCity.ActiveReports.Samples.CustomDataProvider.CsvDataProvider;
using System.Data.Common;

namespace GrapeCity.ActiveReports.Samples.CustomDataProvider.CSVDataProvider
{
	/// <summary>
	/// Implements <see cref="DataProviderFactory"/> for .NET Framework CSV Data Provider.
	/// </summary>
	public class CsvDataProviderFactory : DbProviderFactory
	{
		/// <summary>
		/// Returns a new instance of the <see cref="CsvCommand"/>.
		/// </summary>
		public override DbCommand CreateCommand()
		{
			return new CsvCommand();
		}

		/// <summary>
		/// Returns a new instance of the <see cref="CsvConnection"/>.
		/// </summary>
		public override DbConnection CreateConnection()
		{
			return new CsvConnection();
		}
	}
}
